.github/workflows/commitlint.yml

```
@@ -0,0 +1,33 @@
+name: Commitlint
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
+        with:
+          node-version: '20'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Validate current commit (push)
+        if: github.event_name == 'push'
+        run: npx commitlint --from HEAD~1 --to HEAD --verbose
+
+      - name: Validate PR commits
+        if: github.event_name == 'pull_request'
+        run: npx commitlint --from ${{ github.event.pull_request.base.sha }} --to ${{ github.event.pull_request.head.sha }} --verbose
```

Should we validate every commit in a PR, or focus on ensuring only the final/merge commit follows the convention? This would allow for informal development commits while maintaining clean project history.
```suggestion
    - name: Validate PR commits
      if: github.event_name == 'pull_request'
      run: npx commitlint --from HEAD~1 --to HEAD --verbose
```

---

I think for the PRs, it makes sense to validate all the commits to ensure consistent history and not only the final commit when merging.